File: template.c
#define __GNU_SOURCE
#include <stdio.h>
#include <stdlib.h>


void part1(char* filename);
void part2(char* filename);

int main(int argc, char *argv[])
{
  if (argc != 2) {
    printf("Usage: %s <filename>\n", argv[0]);  
    exit(0);
  }  
  part1(argv[1]);
  part2(argv[1]);
}


void part1(char* filename){
  FILE *file = fopen(filename, "r");

  fclose(file);
}

void part2(char* filename){
  FILE *file = fopen(filename, "r");

  fclose(file);
}

File: template.py
import sys


def main():
    if (len(sys.argv) != 2): 
        print(f"Usage {sys.argv[0]} <filename>")
        return;
    part1(sys.argv[1])
    part2(sys.argv[1])

def part1(filename: str):
    pass


def part2(filename: str):
    pass

if __name__ == '__main__':
    main()
